extensions/mssql/src/features.ts

```diff
@@ -86,7 +96,35 @@ export class AccountFeature implements StaticFeature {
 
 		let params: contracts.RequestSecurityTokenResponse = {
 			accountKey: JSON.stringify(account.key),
-			token: securityToken.token
+			token: securityToken.token,
+			expiresOn: securityToken.expiresOn
+		};
+
+		return params;
+	}
+
+	protected async refreshToken(request: contracts.RefreshTokenParams): Promise<contracts.TokenRefreshedParams> {
+
+		// find account
+		const accountList = await azdata.accounts.getAllAccounts();
+		const account = accountList.find(a => a.key.accountId === request.accountId);
+
+		// find tenant
+		const tenant = account.properties.tenants.find((tenant: azurecore.Tenant) => request.authority.includes(tenant.id));
+		const unauthorizedMessage = localizedConstants.mssqlInsufficentPriveleges(account.key.accountId);
+		if (!tenant) {
+			void window.showErrorMessage(unauthorizedMessage);
```

Comment:
How do we know that the account "didn't have sufficient permissions"? What does that even mean - what permissions did it not have?

It seems like you should just be showing the "couldn't find tenant" error since that at least tells the user what went wrong...